src/main/java/com/shep/fourthlecture/models/base/User.java

```diff
@@ -0,0 +1,46 @@
+package com.shep.fourthlecture.models.base;
+
+import com.shep.fourthlecture.annotations.NotNull;
+import com.shep.fourthlecture.models.impl.*;
+import com.shep.fourthlecture.models.Ticket;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public abstract class User implements Identifiable, Printable {
+    @NotNull
+    private String id;
+    private String role;
+    private String name;
+
+    public User(String id, String role, String name) {
+        this.id = id;
+        this.role = role;
+        this.name = name;
+    }
+
+    public User(String id, String role) {
+        this.id = id;
+        this.role = role;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void printRole() {
+        System.out.println("Role: " + role);
+    }
+
+    public abstract void handleTicket(Ticket ticket);
```

Comment:
"In your application, there should be 2 Users - Client and Admin, both should be able to .printRole() and also have unique functions - getTicket() for User and
checkTicket() - for Admin."

nothing about handleTicket